big_tests/tests/push_integration_SUITE.erl

```diff
@@ -249,17 +250,178 @@ sm_unack_messages_notified_default_plugin(Config) ->
-                             [{body, <<"msg-2">>}, {unread_count, 1}, {badge, 1}]),
+    verify_notification(FCMDevice, <<"fcm">>, [], [{SenderJID, <<"msg-2">>},
+                                                   {BobJID, <<"msg-1">>}]),
 
-    ?assertExit({test_case_failed, _}, wait_for_push_request(FCMDevice, 1000)),
+    ?assertExit({test_case_failed, _}, wait_for_push_request(FCMDevice, 500)),
 
     escalus_connection:stop(Bob).
 
+immediate_notification(Config) ->
+    ConnSteps = [start_stream, stream_features, maybe_use_ssl,
+                 authenticate, bind, session, stream_resumption],
+
+    %% connect bob and alice
+    BobSpec = escalus_fresh:create_fresh_user(Config, bob),
+    {ok, Bob, _} = escalus_connection:start(BobSpec),
+    escalus_connection:send(Bob, escalus_stanza:presence(<<"available">>)),
+    escalus_connection:get_stanza(Bob, presence),
+    BobJID = bare_jid(Bob),
+
+    AliceSpec = [{manual_ack, false}, {stream_management, true} |
+                 escalus_fresh:create_fresh_user(Config, alice)],
+    {ok, Alice, _} = escalus_connection:start(AliceSpec, ConnSteps),
+    escalus_connection:send(Alice, escalus_stanza:presence(<<"available">>)),
+    escalus_connection:get_stanza(Alice, presence),
+
+    #{device_token := APNSDevice} = enable_push_for_user(Alice, <<"apns">>, [], Config),
+    #{device_token := FCMDevice} = enable_push_for_user(Alice, <<"fcm">>, [], Config),
+
+    escalus_connection:send(Bob, escalus_stanza:chat_to(bare_jid(Alice), <<"msg-0">>)),
+    escalus:assert(is_chat_message, [<<"msg-0">>], escalus_connection:get_stanza(Alice, msg)),
+
+    H = escalus_connection:get_sm_h(Alice),
+    escalus:send(Alice, escalus_stanza:sm_ack(H)),
+
+    escalus_connection:send(Bob, escalus_stanza:chat_to(bare_jid(Alice), <<"msg-1">>)),
+    escalus:assert(is_chat_message, [<<"msg-1">>], escalus_connection:get_stanza(Alice, msg)),
+
+    C2SPid = mongoose_helper:get_session_pid(Alice, distributed_helper:mim()),
+    escalus_connection:kill(Alice),
+
+    verify_notification(FCMDevice, <<"fcm">>, [], BobJID, <<"msg-1">>),
+
+    escalus_connection:send(Bob, escalus_stanza:chat_to(bare_jid(Alice), <<"msg-2">>)),
+    verify_notification(FCMDevice, <<"fcm">>, [], BobJID, <<"msg-2">>),
+
+    ?assertExit({test_case_failed, _}, wait_for_push_request(APNSDevice, 500)),
+
+    rpc(?RPC_SPEC, sys, terminate, [C2SPid, normal]),
+
+    verify_notification(APNSDevice, <<"apns">>, [], [{BobJID, <<"msg-1">>},
+                                                     {BobJID, <<"msg-2">>}]),
+
+    ?assertExit({test_case_failed, _}, wait_for_push_request(FCMDevice, 500)),
+
+    escalus_connection:stop(Bob).
+
+double_notification_with_two_sessions_in_resume(Config) ->
+
+%%    This test case serves as a demonstration of doubled push notifications
+%%    which occur with multiple push devices and sessions in resume state
+
+%%    diagram presenting the test's logic
+%%    generated using https://www.sequencediagram.org/
```

Comment:
Thanks a lot for the sequence diagram!

Reply:
i anyway needed it for the design docs :)